orte/mca/rml/base/rml_base_channel_handlers.c

```diff
@@ -0,0 +1,539 @@
+/*
+ *
+ * Copyright (c) 2015 Intel, Inc. All rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+/** @file:
+ *
+ */
+
+/*
+ * includes
+ */
+#include "orte_config.h"
+
+#ifdef HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include "orte/constants.h"
+#include "orte/types.h"
+
+#include "opal/dss/dss.h"
+#include "opal/util/output.h"
+#include "opal/util/timings.h"
+#include "opal/class/opal_list.h"
+
+#include "orte/mca/errmgr/errmgr.h"
+#include "orte/runtime/orte_globals.h"
+#include "orte/runtime/orte_wait.h"
+#include "orte/util/name_fns.h"
+
+#include "orte/mca/rml/rml.h"
+#include "orte/mca/rml/base/base.h"
+#include "orte/mca/rml/base/rml_contact.h"
+#include "orte/mca/qos/base/base.h"
+
+
+static int unpack_channel_attributes (opal_buffer_t *buffer, opal_list_t *qos_attributes);
+static orte_rml_channel_t * get_channel ( orte_process_name_t * peer,
+                                          opal_list_t *qos_attributes,
+                                          bool recv);
+static int send_open_channel_reply (orte_process_name_t *peer,
+                                    orte_rml_channel_t *channel,
+                                    bool accept);
+void orte_rml_base_close_channel(int fd, short flags, void *cbdata)
+{
+    orte_rml_send_request_t *req = (orte_rml_send_request_t*)cbdata;
+    orte_rml_close_channel_t *close_chan;
+    OPAL_OUTPUT_VERBOSE((1, orte_rml_base_framework.framework_output,
+                         "%s rml_close_channel to peer %s",
+                         ORTE_NAME_PRINT(ORTE_PROC_MY_NAME),
+                         ORTE_NAME_PRINT(&req->post.close_channel.channel->peer)));
+    OPAL_TIMING_EVENT((&tm_rml, "to %s", ORTE_NAME_PRINT(&peer)));
```

Comment:
peer is undeclared here. Please compile with --enable-timing option and fix other similar errors.